File: backend/app/schemas/sugerencia.py
from datetime import datetime
from typing import Optional

from pydantic import BaseModel, Field

from app.constants import EstadoSugerencia


class AdjuntoResponse(BaseModel):
    id: str
    nombre: str
    tipo: str
    tamano: int
    url: str


class SugerenciaResponse(BaseModel):
    id: str
    titulo: str
    descripcion: str
    categoria: Optional[str] = None
    firmas: int
    metaFirmas: int
    fechaPublicacion: str
    fechaVencimiento: str
    estado: EstadoSugerencia
    adjuntos: list[AdjuntoResponse] = Field(default_factory=list)


class SugerenciaResumenResponse(BaseModel):
    id: str
    titulo: str
    descripcion: str
    categoria: Optional[str] = None
    firmas: int
    metaFirmas: int
    fechaPublicacion: str
    fechaVencimiento: str
    estado: EstadoSugerencia
    cantidadAdjuntos: int


class MensajeResponse(BaseModel):
    message: str


class ErrorResponse(BaseModel):
    detail: str

File: frontend/src/types/sugerencia.ts
export type EstadoSugerencia = 'ACTIVA' | 'ELEGIBLE' | 'EXPIRADA' | 'ENVIADA';

export interface Adjunto {
  id: string;
  nombre: string;
  tipo: string;
  tamano: number;
  url: string;
}

export interface Sugerencia {
  id: string;
  titulo: string;
  descripcion: string;
  categoria?: string;
  firmas: number;
  metaFirmas: number;
  fechaPublicacion: string;
  fechaVencimiento: string;
  estado: EstadoSugerencia;
  adjuntos: Adjunto[];
}

export interface SugerenciaResumen {
  id: string;
  titulo: string;
  descripcion: string;
  categoria?: string;
  firmas: number;
  metaFirmas: number;
  fechaPublicacion: string;
  fechaVencimiento: string;
  estado: EstadoSugerencia;
  cantidadAdjuntos: number;
}

export interface CrearSugerenciaPayload {
  titulo: string;
  descripcion: string;
  categoria?: string;
  archivos: File[];
}

export interface FiltrosSugerencia {
  busqueda?: string;
  estado?: EstadoSugerencia | 'TODAS';
}
